src/screenfull.js

```diff
@@ -109,9 +109,18 @@
 					resolve();
 				}.bind(this);
 
-				document[fn.exitFullscreen]();
+				if (this.isFullscreen) {
+						return document[fn.exitFullscreen]()
```

Comment:
You should not rely on this being a promise as it's not true for all browsers. Your logic below also makes no sense when you do that as you're essentially doing the same thing twice.